File: src/main/java/me/miki/shindo/management/settings/impl/CategorySetting.java
package me.miki.shindo.management.settings.impl;

import me.miki.shindo.management.settings.config.ConfigOwner;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.management.settings.Setting;

public class CategorySetting extends Setting {

    public CategorySetting(TranslateText text, ConfigOwner parent) {
        super(text, parent);
    }

    public CategorySetting(String name, ConfigOwner parent) {
        super(name, parent);
    }

    private boolean collapsed;

    public boolean isCollapsed() {
        return collapsed;
    }

    public void setCollapsed(boolean collapsed) {
        this.collapsed = collapsed;
    }

    public void toggle() {
        this.collapsed = !this.collapsed;
    }
}

File: src/main/kotlin/me/miki/shindo/management/settings/impl/CategorySetting.kt
package me.miki.shindo.management.settings.impl

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.management.settings.Setting
import me.miki.shindo.management.settings.config.ConfigOwner

open class CategorySetting : Setting {
    private var collapsed = false

    constructor(text: TranslateText, parent: ConfigOwner) : super(text, parent)

    constructor(name: String, parent: ConfigOwner) : super(name, parent)

    fun isCollapsed(): Boolean = collapsed

    fun setCollapsed(collapsed: Boolean) {
        this.collapsed = collapsed
    }

    fun toggle() {
        collapsed = !collapsed
    }
}
